pkg/state/state.go

```diff
@@ -3339,22 +3328,3 @@ func (st *HelmState) getOCIChart(pullChan chan PullCommand, release *ReleaseSpec
 
 	return &chartPath, nil
 }
-
-// Pull charts one by one to prevent concurrent pull problems with Helm
```

Comment:
I'm just curious why this part is no longer needed? does helm fix it upstream?

Reply:
In the worker which manipulate the `pullChan` channel is pulling charts only for the `getOCIChart` function and the chart pulled are never used : this worker runs for nothing. 

I don't know why this code was write but today, if it works, I imagine the previous helm issue you speak about is fixed.

Reply:
@jycamier Can we maintain the original functional logic, reform it, or give a better reason to clean it up?

Reply:
For your reference, here's the original issue that covers the helm pull issue:
https://github.com/roboll/helmfile/issues/1705

Reply:
From my limited test result, I can say that this is no longer an issue since helm 3.8.

Previsouly `helm chart pull` pulled to the chart into a local registry cache maintained by helm and apparently it was subject to a race condition.

However since Helm 3.8 it seems to pull into the local directory- naturally, there will be no place for race conditions as long as you don't pull the same chart multiple times while reading the chart before all other chart pull operations complete. Helmfile does wait for all the chart pull operations to finish before doing subsequent operations like chartify, diff, template, and so on.

That said, I think it's fine to remove this legacy code in this PR.

Reply:
@mumoshu ok. got it. Thanks very much.

Reply:
> I can say that this is no longer an issue since helm 3.8.

Let me correct this. I meant to say "helm 3.7"